spyder/plugins/projects/plugin.py

```diff
@@ -409,6 +433,49 @@ def get_project_types(self):
         """
         return self.get_widget().get_project_types()
 
+    def handle_switcher_modes(self, mode):
```

Comment:
```suggestion
    def _handle_switcher_modes(self, mode):
```

* This follows my suggestion above.
* Also, please move this method and the two others below it to the `Private API` section below.